build-setup/clion/using_clion.md

```diff
@@ -20,164 +16,50 @@ by simply calling `g++` or `clang++`.
-visible and accessible from CLion.
-
-You can enable automatic reload of the project for every change in `compile_commands.json`:
- * In CLion 2018.2 set the **Use auto-import** checkbox in **Settings / Preferences | Build, Execution, Deployment | Compilation Database**.
- * From CLion 2020.2 set both the **Reload project after changes in the build scripts**
- and **Any changes** checkboxes in **Settings / Preferences | Build, Execution, Deployment | Build Tools** 
-
-To follow up the changes in the Makefiles, we can create File Watchers (you should have installed File Watchers plugin).
-Navigate to **Settings / Preferences | Tools | File Watchers** and create a new File Watcher for all files of type **GNU
-Makefile** located in the project root and subdirectories:
-
-**File type:** GNU Makefile\
-**Scope:** Project Files\
-**Program:** `$ProjectFileDir$/dev-tools/compiledb.sh`\
-**Arguments:** `$ProjectFileDir$/compile_commands.json`\
-**Working directory:** `$FileDir$`
-- [x] Auto-save edited files to trigger the watcher
-- [x] Trigger the watcher on external changes
-
 
-## Create custom build targets for building and testing
+## Importing the `ml-cpp` project
 
-The compilation database itself lacks the data required for **building, running and debugging** an application. However,
-you can set up the workflow by adding **custom build targets** for your compilation database project and creating custom
-**Run/Debug configurations** for these targets.
+Open `CLion` and close any existing open project. From the `Welcome to CLion` screen click on `Open` and navigate to
+your `ml-cpp` repo. 
 
-### Custom build targets
+![Open Project](./open_project.png)
 
-Let's create a custom target to build the libraries, which gives you the same behavior as running `make` in the
-project root. Go to **Settings / Preference | Build, Execution, Deployment | Custom Build Targets** and click **+** to
-add a new target. Pick the name, in this tutorial we will use the name *make build*.
+Ensure `CMake project` is selected and click the `OK` button. You will be presented with the `CLion` preferences
+for the `ml-cpp` project.
 
-In the area **Toolchain** we have to specify custom tools for building and cleaning the project.
+![Default Configuration Options](./default_project_config_options.png)
 
-For **Build:** click on **...** to open **External Tools** window and then on **+** to create a new external tool.
+By default `CLion` will opt for generating a `Ninja` build system. While this is a perfectly valid option, for
+better compatibility with building from the command line it is preferred to configure `CLion` to generate `Unix Makefiles`.
+To do this either select `Unix Makefiles` from the `Generator` drop-down or type `-G "Unix Makefiles"` in the
+`CMake options` field. Add `cmake_build` to the `Build directory` field and e.g. `-j 10` to the `Build options` field
```

Comment:
This is an interesting conundrum.

CLion defaults to `cmake-build-release` and `cmake-build-debug` for its CMake build directories.  Everything else we're doing is using `cmake_build`.  I am wondering if we should actually use `cmake-build-release` everywhere where we've used `cmake_build` currently.

If we recommend CLion users hardcode their config to `cmake_build` then when they select a debug build we'll be potentially mixing stuff in the CMake build directory with release builds done from the command line.

I think CLion has actually got it right by selecting different CMake build directories for debug and release to make switching between the two safer.

Do you agree or is there a good argument the other way?

If we think we might ever want to change then now is the time, before people get used to one directory name.  I'm not sure how much work it is to change `cmake_build` to `cmake-build-release` everywhere.  Hopefully with a bit of `find`, `xargs` and `sed` it's not too hard.

The `.gitignore` and `.dockerignore` files should also list both `cmake-build-release` and `cmake-build-debug` if we do this.

Reply:
That's an interesting point, yes I agree that it makes sense to go with the CLion naming scheme (plus its one less thing to configure when getting set up). I'll make the changes now.